### ISSUE TITLE
Tooltip a11y - dialog role needs name

### ISSUE THREAD

**Issue by finken2** [carbon-design-system/carbon#9229]
Tooltip throws accessibility error:
Interactive component with WAI-ARIA role 'dialog' does not have a programmatically associated name

https://github.ibm.com/alchemy-containers/satellite-ui/issues/1384

It's also recreatable in the carbon react storybook:
https://react.carbondesignsystem.com/?path=/story/components-tooltip--default-bottom

===== ENVIRONMENT =====
IBM CI 162 Checkpoint Violation: 4.1.2 Name, Role, Value
IBM Guidance: https://www.ibm.com/able/requirements/requirements/#4_1_2
Test Tool Type: IBM Equal Access Accessibility Checker
Browser: Chrome

=====DEFECT DESCRIPTION=====
===== ENVIRONMENT =====
IBM CI 162 Checkpoint Violation: 4.1.2 Name, Role, Value
IBM Guidance: https://www.ibm.com/able/requirements/requirements/#4_1_2
Test Tool Type: IBM Equal Access Accessibility Checker
Browser: Chrome

=====DEFECT DESCRIPTION=====
Interactive component with WAI-ARIA role 'dialog' does not have a programmatically associated name.

**Comment by tw15egan** [comment]
I'm able to resolve the issue by removing the `tooltipBodyId` from the storybook example, as well as the `<p id="tooltip-body"`, which gets rid of the error

```jsx
<Tooltip {...props.withIcon()} tooltipId="tooltip-body">
      <p>.....
```

Voiceover still seems to work fine, but may be affecting JAWS / other screenreaders. cc @dakahn

**Comment by andreancardona** [comment]
@joshblack just consulted with @dakahn on this ( I think he mentioned it to you). But we are adding this issue to next week's (8/12) Accessibility Guild meeting - [agenda found here](https://ibm.ent.box.com/notes/840018414767)

a current fix that we will present is adding a dynamic `aria-label` to the `tooltip__content` element as seen below
